arcgis-ios-sdk-samples/Layers/Map image layer tables/MapImageLayerTablesViewController.swift

```diff
@@ -0,0 +1,150 @@
+        // Create an action sheet to display the various comments to choose from.
+        let alertController = UIAlertController(title: "Related Service Requests", message: "Select a comment to view related spatial features on the map.", preferredStyle: .actionSheet)
+        // Create an action for each comment.
+        commentsArray.forEach { feature in
+            // Extract the "comments" attribute as a string.
+            let commentsTitle = feature.attributes["comments"] as! String
+            // Create an action with the comments title.
+            let action = UIAlertAction(title: commentsTitle, style: .default) { [weak self] (_) in
+                guard let self = self else { return }
+                // Clear the former graphics.
+                self.selectedFeaturesOverlay.graphics.removeAllObjects()
+                // Get the map image layer that contains the service request sublayer and the service request comments table.
+                let serviceRequestsMapImageLayer = self.mapView.map?.operationalLayers.firstObject as? AGSArcGISMapImageLayer
+                // Get the (non-spatial) table that contains the service request comments.
+                guard let commentsTable = serviceRequestsMapImageLayer?.tables.first else { return }
+                // Get the relationship that defines related service request features for features in the comments table (this is the first and only relationship).
+                let commentsRelationshipInfo = commentsTable.layerInfo?.relationshipInfos.first
+                // Create query parameters to get the related service request for features in the comments table.
+                let relatedQueryParameters = AGSRelatedQueryParameters(relationshipInfo: commentsRelationshipInfo!)
+                relatedQueryParameters.returnGeometry = true
+                // Disable the query button while the feature loads.
+                self.queryBarButtonItem.isEnabled = false
+                self.queryCommentsTable(feature: feature, commentsTable: commentsTable, relatedQueryParameters: relatedQueryParameters)
+            }
+            // Add the action to the controller.
+            alertController.addAction(action)
+        }
+        // Add "cancel" item.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+        // Present the controller.
+        alertController.popoverPresentationController?.barButtonItem = queryBarButtonItem
+        present(alertController, animated: true)
+    }
+    
+    /// The service request URL.
+    static let serviceRequestURL = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/ServiceRequest/MapServer")!
+    /// The map image layer that uses the service URL.
+    let serviceRequestsMapImageLayer = AGSArcGISMapImageLayer(url: serviceRequestURL)
+    /// The array to store the possible comments.
+    var commentsArray = [AGSFeature]()
```

Comment:
I believe Phil once suggested against this style.
```suggestion
    var commentsArray: [AGSFeature] = []
```